wasp/apps/snake.py

```diff
@@ -0,0 +1,219 @@
+    b'$\x816\xc0\x01\xc1@\x03AAQA\x80\x02\x81\x01'
+    b'\xc0\xff\xc1\xd1@IA6\x80\x01\x81\xc0\x03\xc1\xc1\xd1'
+    b'\xc1@\x02A\x01\x80\xb6\x81\xc0\xdb\xc1\xce\xc1\xc1@$'
+    b'A6\x80\x01\x81\xc0\x03\xc1\xd2\xc1@\x02A\x01\x01\x01'
+    b'\x0e\x01\x01\x016\x01\x81\x81\x01\x0e\x01\x01\x81\x01\x14\x01'
+    b'4\x01\x01\x01\x80H\x81\xc0I\xc1\x8e\xc1\x81\x01\x01\x01'
+    b'\xc1\xc1\xce\xc1\xc1\x017\x01@\xffAQ\x80$\x81\x01'
+    b'\x01AQ\x817\x01AANAA\x81\x01\x01AA'
+    b'NAA\x817\x01\xc0\xdf\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01\xc1ANAA\x81\x01\x01'
+    b'\xc1ANAA\x817\x01AQ@IA\x01\x01\x80'
+    b'\xff\x81\x91A7\x01\xc0\xb6\xc1@\xdbANAA\x80'
+    b'$\x81\x01\x01\xc1ANAA\x818\x12\x03\x12?\r'
+    b'\x01\x14\x01?\x9c'
+)
+
+
+import wasp, time
+from random import randint
+from apps.launcher import LauncherApp
+
+class SnakeGameApp():
+    NAME = 'Snake'
+    ICON = snake_game
+
+    def __init__(self):
+        self.running = True
+        self.snake = Snake()
+        self.foodLocation()
+        self.highscore = 1
+        
+
+    def foreground(self):
+        """Activate the application."""
+        wasp.watch.drawable.fill()
+        self._draw()
+        wasp.system.request_event(wasp.EventMask.TOUCH |
+                                  wasp.EventMask.SWIPE_UPDOWN |
+                                  wasp.EventMask.SWIPE_LEFTRIGHT)
+        wasp.system.request_tick(250)
+
+    def background(self):
+        """De-activate the application."""
+        pass
+
+    def sleep(self):
+        """Notify the application the device is about to sleep."""
+        return False
+
+    def wake(self):
+        """Notify the application the device is waking up."""
+        pass
+    
+    def touch(self,event):
+        if not self.running:
+            self.running = True
+            wasp.watch.drawable.fill()
+
+    def swipe(self, event):
+      if self.running:
+        """Notify the application of a touchscreen swipe event."""
+        if event[0] == wasp.EventType.UP:
+            self.snake.setDir(0,-15)
+        elif event[0] == wasp.EventType.DOWN:
+            self.snake.setDir(0,15)
+        elif event[0] == wasp.EventType.LEFT:
+            self.snake.setDir(-15,0)
+        elif event[0] == wasp.EventType.RIGHT:
+            self.snake.setDir(15,0)
+        else:
+            print("Error!")
+      else:
+        self.running = False
+        wasp.system.switch(LauncherApp())
```

Comment:
No need to open code this. If you want to allow the swipe to work "as normal" when the game isn't running just `return True` from the swipe() funciton and the event will be passed to the system event handling.